clients/da-vinci-client/src/main/java/com/linkedin/davinci/store/rocksdb/RocksDBStoragePartition.java

```diff
@@ -548,6 +552,79 @@ public byte[] get(ByteBuffer keyBuffer) {
     }
   }
 
+  public List<byte[]> multiGet(List<byte[]> keys) {
+    readCloseRWLock.readLock().lock();
+    try {
+      makeSureRocksDBIsStillOpen();
+      return rocksDB.multiGetAsList(keys);
+    } catch (RocksDBException e) {
+      throw new VeniceException("Failed to get value from store: " + storeName + ", partition id: " + partitionId, e);
+    } finally {
+      readCloseRWLock.readLock().unlock();
+    }
+  }
+
+  public List<ByteBuffer> multiGet(List<ByteBuffer> keys, List<ByteBuffer> values) {
+    readCloseRWLock.readLock().lock();
+    try {
+      makeSureRocksDBIsStillOpen();
+      List<ByteBufferGetStatus> statusList = rocksDB.multiGetByteBuffers(keys, values);
+
+      List<ByteBuffer> resultList = new ArrayList<>(keys.size());
```

Comment:
It should be ok to reuse `values` and avoid excessive allocation, at least in the scenario when value buffers are sized probably. But even in general case it might be ok to modify `values` list since most likely the caller allocates a new list each time it calls the api.

Reply:
The passed `values` param is for reuse purpose, so I don't think customer will allocate a new list every time and that was also the reason to construct a new list, which should be cheap since all the elements are referring to the populated buffer.